pre_commit/commands/run.py

```diff
@@ -417,7 +418,17 @@ def run(
         to_install = [hook for hook in hooks if hook.id not in skips]
         install_hook_envs(to_install, store)
 
-        retval = _run_hooks(config, hooks, skips, args, environ)
+        if editor.should_clean_draft(args.hook_stage):
+            editor.clean_draft()
+            logger.info(f'Moved outdated commit message to `{editor.COMMIT_MESSAGE_EXPIRED_DRAFT_PATH}`.')
+
+        if editor.should_run_concurrently(args.hook_stage):
+            retval = editor.run_concurrently(_run_hooks, config, hooks, skips, args, environ)
```

Comment:
It might be a bit cleaner to just have this take a no-args callable. Then we could define a lambda here and either pass it in or just invoke it below. Also gets rid of some `Any`s we've got.

Reply:
After some consideration, I actually think that's not a good change. Python's lambdas/defs are _not_ proper closures -- they use value of the variables they reference _at call time_, not when they're defined. This can lead to weird, subtle bugs and is best avoided, IMO, except in simple cases.  E.g.:
```python
# Python 3.7.9 (default, Jan 24 2021, 17:48:25)
>>> x = 1
>>> def f():
...   print(x)
...
>>> f()
1
>>> x = 2
>>> f()
2

>>> x = 1
>>> l = lambda: x
>>> l()
1
>>> x = 2
>>> l()
2
```